jobdispatcher/build.gradle

```diff
@@ -20,21 +25,183 @@ android {
+    group "artifact"
+    description "Builds the library AARs"
+}
+
+def thickDesc = "The Firebase JobDispatcher is a library that provides a common" +
+    " interface to scheduling engines on Android."
+def thinDesc = thickDesc + " This version should be used by anyone with a" +
+    " dependency on the legacy GCM library (com.google.android.gms:play-services-gcm)."
+
+def flavorsToPublicationsMap = [
+    thick: [
+        description: thickDesc,
+        artifactId: "firebase-jobdispatcher-android",
+    ],
+    thin: [
+        description: thinDesc,
+        artifactId: "firebase-jobdispatcher-android-thin",
+    ],
+]
+
+publishing {
+    repositories {
+        // build repo for quick validation
+        maven {
+            name "buildLocal"
+            url "${buildDir}/repo"
+        }
+    }
+
+    publications {
+        android.productFlavors.each { flavor ->
+            "dispatcher${flavor.name.capitalize()}Library"(MavenPublication) {
+                artifactId flavorsToPublicationsMap[flavor.name].artifactId
+                groupId project.ext.group
+                version project.ext.version
+
+                artifact("${buildDir}/outputs/aar/jobdispatcher-${flavor.name}-release.aar") {
+                    // wrap in a closure because this task isn't defined 'till
+                    // the android plugin finishes
+                    builtBy { tasks["bundle${flavor.name.capitalize()}Release"] }
+                }
+                artifact sourcesJar
+                artifact javadocsJar
+
+                pom.withXml {
+                    def n = asNode()
+
+                    n.appendNode("description", flavorsToPublicationsMap[flavor.name].description)
+                    n.appendNode("inceptionYear", 2016)
+
+                    def licensesNode = n.appendNode("licenses")
+                    def licenseNode = licensesNode.appendNode("license")
+                    licenseNode.appendNode("name", project.ext.licenseName)
+                    licenseNode.appendNode("url", project.ext.licenseUrl)
+                    licenseNode.appendNode("distribution", project.ext.licenseDist)
+
+                    def developersNode = n.appendNode("developers")
+                    def devNode = developersNode.appendNode("developer")
+                    devNode.appendNode("id", "firebase")
+                    devNode.appendNode("name", "Firebase")
+                    devNode.appendNode("email", "support@firebase.com")
+                    devNode.appendNode("url", "https://firebase.com/")
+                    devNode.appendNode("organization", "Firebase")
+                    devNode.appendNode("organizationUrl", "https://firebase.com/")
+                    def rolesNode = devNode.appendNode("roles")
+                    rolesNode.appendNode("role", "architect")
+                    rolesNode.appendNode("role", "developer")
+
+                    devNode.appendNode("timezone", "-8" /* Hi from MTV */)
```

Comment:
This seems terrible as you'd need to change it for DST, is it used for anything important?